Documentation/CMake.md

```diff
@@ -59,21 +59,34 @@ endif()
 ## Add an entry point
 
 You must include a source file in your test executable target with a
-`@main` entry point. The following example uses the SwiftPM entry point:
+`@main` entry point. The example main below requires the experimental
+`Extern` feature. The declaration of `swt_abiv0_getEntryPoint` could
+also be written in a C header file with its own `module.modulemap`.
 
 ```swift
-import Testing
+typealias EntryPoint = @convention(thin) @Sendable (_ configurationJSON: UnsafeRawBufferPointer?, _ recordHandler: @escaping @Sendable (_ recordJSON: UnsafeRawBufferPointer) -> Void) async throws -> Bool
+
+@_extern(c, "swt_abiv0_getEntryPoint")
+func swt_abiv0_getEntryPoint() -> UnsafeRawPointer
 
 @main struct Runner {
-    static func main() async {
-        await Testing.__swiftPMEntryPoint() as Never
+    static func main() async throws {
+        nonisolated(unsafe) let configurationJSON: UnsafeRawBufferPointer? = nil
+        let recordHandler: @Sendable (UnsafeRawBufferPointer) -> Void = { _ in }
+
+        let entryPoint = unsafeBitCast(swt_abiv0_getEntryPoint(), to: EntryPoint.self)
+
+        if try await entryPoint(configurationJSON, recordHandler) {
+            exit(EXIT_SUCCESS)
+        } else {
+            exit(EXIT_FAILURE)
+        }
```

Comment:
The entry point function itself will only throw an error if some problem was encountered during the act of running the tests. (An example of this might be a system error encountered while discovering or planning the set of tests to run.) It will _not_ throw an error if a test successfully ran, but failed.

Typically, a test runner executable is expected to exit with a nonzero status if at least one test failed, so ideally it would be better for this logic to implement a real record handler, interpret the records it receives, detect if one represents an issue, and set a flag whenever at least one issue is received so it can `exit(EXIT_FAILURE)`.

I realize that's more complicated to document here, and we don't necessarily need to go that far in this PR. But I do think it would be helpful to at least leave a code comment or something indicating that a well-behaved test runner executable should implement such logic.

Reply:
It does return `false` on failure-during-testing though, if any unknown issues were recorded. :)

Reply:
Great, that addresses my concern